hbase-common/src/main/java/org/apache/hadoop/hbase/nio/ByteBuff.java

```diff
@@ -78,6 +79,14 @@ public boolean release() {
     return refCnt.release();
   }
 
+  public RefCnt getRefCnt() {
+    return this.refCnt;
+  }
+
+  public void setRefCnt(RefCnt refCnt) {
```

Comment:
I think setRefCnt for a given ByteBuff is dangerous ?  because its previous recycler will be lost, so better not to expose this method as public ? We have one similar issue before, see: 
```
  /**
   * In theory, the upstream should never construct an ByteBuff by passing an given refCnt, so
   * please don't use this public method in other place. Make the method public here because the
   * BucketEntry#wrapAsCacheable in hbase-server module will use its own refCnt and ByteBuffers from
   * IOEngine to composite an HFileBlock's ByteBuff, we didn't find a better way so keep the public
   * way here.
   */
  public static ByteBuff wrap(ByteBuffer[] buffers, RefCnt refCnt) {
```

Reply:
have not find a better way yet, any suggestions?